### PR TITLE
Remove the avformat_find_stream_info call from the video loader when not needed

### PR DESCRIPTION
**Pull request by JanuszL**
- video reader calls avformat_find_stream_info to obtain video properties.
  In most formats header provides them and there is no need to call
  mentioned function which attempts to decode a couple of the video
  initial frames.

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- video reader calls avformat_find_stream_info to obtain video properties.
  In most formats header provides them and there is no need to call
  mentioned function which attempts to decode a couple of the video
  initial frames.

Relates to https://github.com/NVIDIA/DALI/issues/5040

### Affected modules and functionalities:
- video loader

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  -  test_video.py
  - test_video_pipeline.py
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/reader/loader/video_loader.cc [hunk 1]
```diff
@@ -328,6 +321,18 @@ VideoFile& VideoLoader::get_or_open_file(const std::string &filename) {
     int width = codecpar(stream)->width;
     int height = codecpar(stream)->height;
     auto codec_id = codecpar(stream)->codec_id;
+
+    if (width == 0 || height == 0) {
```
Comment (JanuszL) [inline dali/operators/reader/loader/video_loader.cc:325]: We do the same as in https://github.com/NVIDIA/DALI/blob/main/dali/operators/reader/loader/video/frames_decoder.cc#L160.

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/nvdecoder/nvdecoder.cc [hunk 2]
```diff
@@ -197,6 +197,9 @@ int NvDecoder::handle_display(void* user_data, CUVIDPARSERDISPINFO* disp_info) {
 int NvDecoder::handle_sequence_(CUVIDEOFORMAT* format) {
   int ret = kNvcuvid_failure;
   try {
+    // utilize the NVDEC parser to determine if the video is full range
+    recv_queue_.front().full_range =
+        static_cast<bool>(format->video_signal_description.video_full_range_flag);
```
Comment (stiepan) [inline dali/operators/reader/nvdecoder/nvdecoder.cc:202]: Does it replace the deleted video_loader.cc:516? Why is this better?

**Review by stiepan — APPROVED** [review]

File: dali/operators/reader/nvdecoder/nvdecoder.cc (hunk 2 above)
Comment (stiepan) [reply dali/operators/reader/nvdecoder/nvdecoder.cc:202]: Makes sense, thanks!

**Review by szalpal — APPROVED** [review]

**Pull request merged**